pkg/fakerp/fakerp.go

```diff
@@ -77,11 +80,39 @@ func createOrUpdate(ctx context.Context, log *logrus.Entry, cs, oldCs *api.OpenS
 	// API representation for comparison.
 	if !shared.IsUpdate() {
 		// If containerservice.yaml does not exist - it is Create call
-		// create DNS records only on first call
-		err := CreateOCPDNS(ctx, os.Getenv("AZURE_SUBSCRIPTION_ID"), os.Getenv("RESOURCEGROUP"), cs.Location, os.Getenv("DNS_RESOURCEGROUP"), os.Getenv("DNS_DOMAIN"), os.Getenv("NOGROUPTAGS") == "true")
-		if err != nil {
-			return nil, err
+		// create DNS records only on first call and when user does not supply them
+		if cs.Properties.RouterProfiles[0].PublicSubdomain == "" {
+			log.Info("routeprofiles[0].publicsubdomain was empty, randomizing publicSubdomain and routerCName")
+
+			// generate a random domain string
+			rDomain, err := utilrs.RandomString("abcdefghijklmnopqrstuvxyz0123456789", 20)
+			if err != nil {
+				return nil, err
+			}
+			publicSubdomain = fmt.Sprintf("%s.%s", rDomain, os.Getenv("DNS_DOMAIN"))
+
+			rrCName, err := utilrs.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 20)
+			if err != nil {
+				return nil, err
+			}
+			// create router wildcard DNS CName record
+			routerCName = fmt.Sprintf("osa%s.%s.%s", rrCName, cs.Location, "cloudapp.azure.com")
+
+			log.Info("routeprofiles[0].publicsubdomain was empty, creating dns")
+			err = CreateOCPDNS(ctx, os.Getenv("AZURE_SUBSCRIPTION_ID"), os.Getenv("RESOURCEGROUP"), cs.Location, os.Getenv("DNS_RESOURCEGROUP"), os.Getenv("DNS_DOMAIN"), publicSubdomain, routerCName, os.Getenv("NOGROUPTAGS") == "true")
```

Comment:
Currently our tests fail if we move the DNS creation until after the cluster is created.  The initial health check that determines a cluster is healthy polls indefinitely.  I would like to move this to after cluster creation ~line 180.  I think this will be possible after the certificate generation matches the names for the cluster.
```
INFO[2019-01-30T11:32:04-05:00]/home/kwoodson/go/src/github.com/openshift/openshift-azure/pkg/util/wait/wait.go:68 github.com/openshift/openshift-azure/pkg/util/wait.forHTTPStatusOk.func1() Get https://kwoodsontest1.eastus.cloudapp.azure.com/healthz: x509: certificate signed by unknown authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "openshift-signer") 
```

Or should it work now and I've made a poor assumption?

Reply:
Router DNS is created during cluster bootstrap. Same in real and fakeRP.
Why do we want to move DNS creation after a cluster is created? I might miss something. RealRP creates it before, and I think we should too? :man_shrugging:

Reply:
@mjudeikis, @jim-minter and I had a discussion and I think we wanted to mimic the real RP and not have our vanity DNS setup before the cluster creation.  If we setup DNS post cluster create then we would get a realistic test that we are using the correct internal URLs for health checks.  This uncovered the admin console issue where we are not able to connect to it with our current URL until DNS is correctly set.